src/Raven.Server/Documents/Document.cs

```diff
@@ -70,6 +70,14 @@ public void EnsureMetadata()
             if (IndexScore.HasValue)
                 mutatedMetadata[Constants.Documents.Metadata.IndexScore] = IndexScore;
 
+            if (NonPersistentFlags.HasFlag(NonPersistentDocumentFlags.FromSmuggler) &&
```

Comment:
This should be handled in StreamSource.GetDocuments.
We have a BlittableMetadataModifier there and you set a mode there that will strip the `@counters` property from the metadata when the document is being read.

Reply:
done